contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -274,6 +279,329 @@ transform_funcexpr(Node *node)
  * Case 3: if the pattern doesn't have a constant prefix
  *		 col LIKE PATTERN -> col ILIKE PATTERN
  */
+
+static Node *
+transform_from_ci_as(Node *node, OpExpr *op, like_ilike_info_t like_entry, coll_info_t coll_info_of_inputcollid)
+{
+	Node	   *leftop = (Node *) linitial(op->args);
+	Node	   *rightop = (Node *) lsecond(op->args);
+	Oid			ltypeId = exprType(leftop);
+	Oid			rtypeId = exprType(rightop);
+	char	   *op_str;
+	Node	   *ret;
+	Const	   *patt;
+	Const	   *prefix;
+	Operator	optup;
+	Pattern_Prefix_Status pstatus;
+	int			collidx_of_cs_as;
+
+	tsql_get_server_collation_oid_internal(true);
+
+	if (!OidIsValid(server_collation_oid))
+		return node;
+
+	/*
+	 * Find the CS_AS collation corresponding to the CI_AS collation
+	 * Change the collation of the ILIKE op to the CS_AS collation
+	 */
+	collidx_of_cs_as =
+		tsql_find_cs_as_collation_internal(
+											tsql_find_collation_internal(coll_info_of_inputcollid.collname));
+
+
+	/*
+	 * A CS_AS collation should always exist unless a Babelfish CS_AS
+	 * collation was dropped or the lookup tables were not defined in
+	 * lexicographic order.  Program defensively here and just do no
```

Comment:
Why defensive? Sending the right feedback-error message to the customer would be ideal